massiv/src/Data/Massiv/Array/Manifest/Storable.hs

```diff
@@ -30,21 +32,30 @@ import           Data.Massiv.Array.Unsafe            (unsafeGenerateArray,
                                                       unsafeGenerateArrayP)
 import           Data.Massiv.Core.Common
 import           Data.Massiv.Core.List
+import           Data.Validity
 import qualified Data.Vector.Storable                as VS
 import qualified Data.Vector.Storable.Mutable        as MVS
 import           Foreign.Ptr
 import           GHC.Exts                            as GHC (IsList (..))
+import           GHC.Generics (Generic)
 import           Prelude                             hiding (mapM)
 
 -- | Representation for `Storable` elements
-data S = S deriving Show
+data S = S deriving (Show, Generic)
+
+instance Validity S
 
 type instance EltRepr S ix = M
 
+instance (MVS.Storable e, Validity e) => Validity (VS.Vector e) where
```

Comment:
I wonder why `validity-vector` doesn't have an instance for Storable (and other) vectors, it would definitely make more sense there.

Reply:
Coming soon.

Reply:
I also made that PR to validity.
Now that it's launched, I'll remove the relevant validity instances and update this PR.